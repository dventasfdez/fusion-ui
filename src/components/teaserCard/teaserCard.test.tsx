import React from 'react';
import renderer from 'react-test-renderer';
import TeaserCard, {TeaserCardImg, TeaserCardTop, TeaserCardHeader, TeaserCardBody, TeaserCardFooter} from './teaserCard';

const teaserCardExample = (props?: any) => (
  <TeaserCard {...props}>
    <TeaserCardImg>
      <img
        // eslint-disable-next-line max-len
        src="https://img.freepik.com/fotos-premium/edificios-modernos-torre-o-rascacielos-distrito-financiero-nubes-dia-soleado-chicago-ee-uu_43552-32.jpg?w=2000"
        alt="img-top"
      />
    </TeaserCardImg>
    <TeaserCardTop>
      <div className="status-tag">Semantic</div>
      <span className="tag">{'06 SEP 2022'}</span>
    </TeaserCardTop>
    <TeaserCardHeader>
      <h4>This is a title</h4>
      <span className="subtitle">And this is a subtitle</span>
    </TeaserCardHeader>
    <TeaserCardBody>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, molestias ipsa quis, eum officia aliquid eaque cum
      illum voluptates nobis libero deserunt labore voluptas adipisci minima vero voluptatem veritatis. Rerum?
    </TeaserCardBody>
    <TeaserCardFooter>
      <button className="button_large">Join our newsletter</button>
    </TeaserCardFooter>
  </TeaserCard>
);

/**
 * TEASER CARD
 */
describe('Teaser card type tests', () => {
  test('Teaser card component should render', () => {
    const component = renderer.create(teaserCardExample({className: .stepone-ui'}));
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Teaser card selected component should render', () => {
    const component = renderer.create(teaserCardExample({selected: true}));
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
