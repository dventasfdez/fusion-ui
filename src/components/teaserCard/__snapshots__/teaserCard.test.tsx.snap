// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teaser card snapshots Teaser card component should render 1`] = `
<div>
  <div
    class="card_teaser "
  >
    <div
      class="card-img "
    >
      <img
        alt="img-top"
        src="https://img.freepik.com/fotos-premium/edificios-modernos-torre-o-rascacielos-distrito-financiero-nubes-dia-soleado-chicago-ee-uu_43552-32.jpg?w=2000"
      />
    </div>
    <div
      class="card-top "
    >
      <div
        class="status-tag"
      >
        Semantic
      </div>
      <span
        class="tag"
      >
        06 SEP 2022
      </span>
    </div>
    <div
      class="card-header "
    >
      <h4>
        This is a title
      </h4>
      <span
        class="subtitle"
      >
        And this is a subtitle
      </span>
    </div>
    <div
      class="card-body "
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, molestias ipsa quis, eum officia aliquid eaque cum illum voluptates nobis libero deserunt labore voluptas adipisci minima vero voluptatem veritatis. Rerum?
    </div>
    <div
      class="card-footer "
    >
      <button
        class="button_large"
      >
        Join our newsletter
      </button>
    </div>
  </div>
</div>
`;
